front/components/agent_builder/AgentBuilderFormContext.tsx

```diff
@@ -213,4 +248,35 @@ export type AgentBuilderDataVizAction = z.infer<
   typeof dataVisualizationActionSchema
 >;
 
+export const mcpFormConfigurationSchema = z.object({
+  name: z.string(),
+  description: z.string(),
+  configuration: mcpServerConfigurationSchema,
+});
+
+export type MCPFormConfigurationData = z.infer<
+  typeof mcpFormConfigurationSchema
+>;
+
 export type BaseActionData = z.infer<typeof baseActionSchema>;
+
+// TODO: create types from schema
```

Comment:
I spent enough type to do this but couldn't make it and I gave up for now, will try again tomorrow 🥲